### PR TITLE
Stack Recipe Post Build

### PR DESCRIPTION
**Pull request by ghost**
Adds post_build() to stack recipes, makes cable cuffs, windoors and windows use it
Also nuked the retarded /unanchored window subtypes that caused issues with SMOOTH_TRUE

:cl:
fix: Manually built windows should smooth properly now
/:cl:

credit to tigercat2000 from Paradise

### CODE REVIEW

File: _maps/RandomZLevels/snowdin.dmm [hunk 1]
```diff
@@ -7488,7 +7488,9 @@
 /area/awaymission/snowdin/cave/cavern)
 "rW" = (
 /obj/structure/lattice/catwalk,
-/obj/structure/window/plasma/reinforced/unanchored,
+/obj/structure/window/plasma/reinforced{
+	anchored = 0
```
Comment (Jalleo) [inline _maps/RandomZLevels/snowdin.dmm]: @ShizCalev

**Comment by optimumtact** [comment]
you're undoing work that was done to remove var edits on maps

**Comment by optimumtact** [comment]
I don't know, it is my personal preference that the type system is not a database of var edits.

It causes compiler overhead and costs performance and I don't see the value in it.

However, we have never hashed this out between us as maintainers so I guess now is a good time as any

@tgstation/commit-access

**Comment by ninjanomnom** [comment]
What's it breaking? Most things should be using istype to check typepaths.

Before now I would have said that some types easing the work of mapping is a good thing but we've recently started getting more work done on a better mapping client. With the possibility of better mapping tools there I agree with oranges that we should be moving away from this sort of thing.

**Comment by uraniummeltdown** [comment]
@ninjanomnom it breaks smoothing, most windows use SMOOTH_TRUE where it will only smooth with the exact type paths defined in CanSmoothWith. Some windows like shuttle windows smooth fine regardless since they have that var set to null
But to see this in action, make a window/r-window/plasma window manually by using glass attack self, screw 2 of those windows down next to each other and see that they dont smooth at all since they cannot smooth with unanchored subtypes
the snowdin map is the only map we use these subtypes on

File: code/game/objects/items/stacks/stack_recipe.dm [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	var/res_amount = 1
+	var/max_res_amount = 1
+	var/time = 0
+	var/one_per_turf = FALSE
+	var/on_floor = FALSE
+	var/window_checks = FALSE
+	var/placement_checks = FALSE
+
+/datum/stack_recipe/New(title, result_type, req_amount = 1, res_amount = 1, max_res_amount = 1, time = 0, one_per_turf = FALSE, on_floor = FALSE, window_checks = FALSE, placement_checks = FALSE)
+	src.title = title
```
Comment (Jalleo) [inline code/game/objects/items/stacks/stack_recipe.dm]: remove the unnecessary src

**Comment by uraniummeltdown** [comment]
Just realized that the snowdin windows are corner windows. I can leave in the snowdin window/plasma/reinforced/unanchored subtypes if maints want.
The main purpose of this PR is to get rid of the snowflakiness in stack code /post_build()/ and to make manually built windows smooth correctly @ShizCalev @ninjanomnom

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 3]
```diff
@@ -9,8 +9,8 @@
  * Glass sheets
  */
 GLOBAL_LIST_INIT(glass_recipes, list ( \
-	new/datum/stack_recipe("directional window", /obj/structure/window/unanchored, time = 0, on_floor = TRUE, window_checks = TRUE), \
-	new/datum/stack_recipe("fulltile window", /obj/structure/window/fulltile/unanchored, 2, time = 0, on_floor = TRUE, window_checks = TRUE) \
+	new/datum/stack_recipe/window("directional window", /obj/structure/window, time = 0, on_floor = TRUE, window_checks = TRUE), \
```
Comment (Jalleo) [inline code/game/objects/items/stacks/sheets/glass.dm]: Okay apparently if you change the /obj/structure/window/Initialize(mapload, direct) to have a third parameter you can set the anchored to normally TRUE and override it here to FALSE. I presume you would need to setup a spawner for the corner of the tile windows to do this too and probably have a fourth parameter to set their dir too.

Secondly I was incorrect in regards to src with your datums that is apparently fine.

**Comment by optimumtact** [comment]
lesson learned about smashing unrelated changes together, I suggest moving the type changes to a different pr

**Comment by uraniummeltdown** [comment]
@optimumtact ok will do that soon

**Comment by uraniummeltdown** [comment]
ayy im walkin here

**Pull request closed without merging**